xdsl/rewriter.py

```diff
@@ -119,6 +119,52 @@ def replace_op(
 
         block.erase_op(op, safe_erase=safe_erase)
 
+    @staticmethod
+    def update_op(
+        op: Operation,
+        new_results: Sequence[SSAValue | None] | None = None,
+        safe_erase: bool = True,
+        *,
+        operands: Sequence[SSAValue] | None = None,
+        result_types: Sequence[Attribute] | None = None,
+        properties: Mapping[str, Attribute] | None = None,
+        attributes: Mapping[str, Attribute] | None = None,
+        successors: Sequence[Block] | None = None,
+        regions: Sequence[Region] | None = None,
+    ):
+        """
+        Recreate an operation with any passed field updated. All other fields
+        are take from the original operation.
+
+        If all passed fields are already equals to the existing ones, also just skip and
+        prevent endless pattern recursion.
+        """
+        if (
+            (operands is None or operands == op.operands)
+            and (result_types is None or result_types == op.result_types)
+            and (properties is None or properties == op.properties)
+            and (attributes is None or attributes == op.attributes)
+            and (successors is None or successors == op.successors)
+            and (regions is None or regions == op.regions)
+        ):
+            return False
+        new_op = op.create(
+            operands=op.operands if operands is None else operands,
+            result_types=op.result_types if result_types is None else result_types,
+            # No need to copy the properties, attributes, regions: we replace the
+            # original op so it won't use them anymore!
+            properties=op.properties if properties is None else properties,
+            attributes=op.attributes if attributes is None else attributes,
+            successors=op.successors if successors is None else successors,
+            regions=(
+                tuple(op.detach_region(0) for _ in op.regions)
```

Comment:
```suggestion
                tuple(region.detach() for region in list(op.regions))
```
Makes it more clear to me?